### PR TITLE
SALTO-3656 Fix field parents resolving

### PR DESCRIPTION
**Pull request by dantal4**
Today we assume that if we got a field and its parent type in the same resolve call, the field's parent will point to the same object type that we got to resolve, so we don't need to resolve both.

The problem is that we clone the elements before we resolve them, so the field’s parent don’t point to the same object type that we got to resolve.

---

_Additional context for reviewer_

---
_Release Notes_: 
Workspace:
- Fix field parents resolving

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/workspace/src/expressions.ts [hunk 1]
```diff
@@ -247,12 +247,44 @@ const getResolveFunctions = ({
   }
 }
 
+const getClonedElements = (elements: Element[]): Element[] => {
+  const [fields, restOfElements] = _.partition(elements, isField)
+  const clonedRestOfElements = restOfElements.map(e => e.clone())
+  const clonedTypesMap = new Map(
```
Comment (adamro) [inline packages/workspace/src/expressions.ts:253]: Style question - Do you have a reason to use Map here and not Object.fromEntries?

File: packages/workspace/src/expressions.ts [hunk 2]
```diff
@@ -247,12 +247,44 @@ const getResolveFunctions = ({
...
+      .filter(parent => !clonedTypesMap.has(parent.elemID.getFullName()))
+      .map(parent => [parent.elemID.getFullName(), parent.clone()])
+  )
+  // We want to get the fields from the cloned parents so we keep them pointing the parents
+  // that will be resolved and we don't use an uncloned parent type.
+  const clonedFields = fields.map(field => {
+    const fieldFromClonedParent = clonedTypesMap.get(field.parent.elemID.getFullName())?.fields[field.name]
+      ?? clonedMissingParentsMap.get(field.parent.elemID.getFullName())?.fields[field.name]
+    if (fieldFromClonedParent === undefined) {
+      log.error(
```
Comment (adamro) [inline packages/workspace/src/expressions.ts:268]: Shouldn't this be a warning if you are not actually crashing?

**Review by adamro — APPROVED** [review]
🔥

File: packages/workspace/src/expressions.ts (hunk 1 above)
Comment (dantal4) [reply packages/workspace/src/expressions.ts:253]: I like the typesafety of `map.get` that returns `T | undefined`

File: packages/workspace/src/expressions.ts (hunk 2 above)
Comment (dantal4) [reply packages/workspace/src/expressions.ts:268]: Will change 👍

**Pull request merged**